remotemonitoring/armtemplates/basic-static-map.json

```diff
@@ -584,7 +591,6 @@
             "location": "[variables('location')]",
             "dependsOn": [
                 "[resourceId('Microsoft.Compute/virtualMachines/extensions', parameters('vmName'), 'DockerExtension')]",
-                "[resourceId('Microsoft.Devices/Iothubs/', parameters('iotHubName'))]",
```

Comment:
This `dependsOn` entry needs to be conditional (the dependency doesn't exist if the Hub is already deployed outside this template). Is there a way conditionally have the list entry? I couldn't find anything in the [reference](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-functions).

Reply:
@tiagoshibata Please take a look at this for conditional statements within ARM template. https://azure.microsoft.com/en-us/blog/create-flexible-arm-templates-using-conditions-and-logical-functions/